src/frontend/src/index.ts

```diff
@@ -69,14 +71,15 @@ const init = async () => {
     // Open the management page
     case "manage": {
       // Go through the login flow, potentially creating an anchor.
-      const { userNumber, connection: authenticatedConnection } = await login(
-        connection
-      );
+      const { userNumber, connection: authenticatedConnection } = returning()
+        ? await auth(connection)
```

Comment:
nth: not sure if following does not generate a short js?

```
const fnc = returning() ? auth : welcome;
const { userNumber, connection: authenticatedConnection } = await fnc();
```

not sure typescript would be happy with though...

Reply:
I could also pull out the `await` if it's a matter to make it shorter but I thought it was clearer like this?

Reply:
Not really to because of the repetition but can definitely live with.

note: I tried and answered my question, indeed the js would be shorter

source

```

const a = async () => console.log('a');
const b = async () => console.log('b');

export const text = async (tmp: boolean) => {
    const fnc = tmp ? a : b;
    const c = await fnc();   
}

    // vs

export const text = async (tmp: boolean) => {
    const c = tmp ? await a() : await b();
}
```

generates

```
var a=async()=>console.log("a"),c=async()=>console.log("b"),s=async o=>{let n=o?await a():await c()};export{s as text};

var c=async()=>console.log("a"),n=async()=>console.log("b"),a=async o=>{let t=await(o?c:n)()};export{a as text};
```

few bytes saved 😂